### DISCUSSION TITLE
How to reset FileInput

### DISCUSSION THREAD

**Discussion by raminrad** [MetroStar/comet#262] [Q&A] [answered]
I would like to clear **_FileInput_** after the upload is submitted. How can I do that? Thanks.

**Comment by jbouder** [comment, marked as answer]
Hi @raminrad , I couldn't find a simple way of doing that actually, so I dug a bit into the USWDS implementation of the file input here: https://github.com/uswds/uswds/blob/develop/packages/usa-file-input/src/index.js.

It doesn't look like they provide a simple way either, so I went ahead and replicated how they clear the field, based on the user changing the file selections, and ended up create a new custom hook. Here are the details of the current draft PR: https://github.com/MetroStar/comet/pull/263, including a new story documenting how to make use of this new hook.

Prior to merging this, I went ahead and published the following beta version: `@metrostar/comet-uswds@3.3.0-beta.0`. Can you verify this works for you and get back to me?

**Comment by raminrad** [comment]
@jbouder Thanks so much... just tested the beta version and works like a charm. Thanks.

**Reply by jbouder** [reply]
> @jbouder Thanks so much... just tested the beta version and works like a charm. Thanks.

Awesome! I'll work to get a full release out. Thanks for posting the question! And let me know if anything else pops. up.

**Reply by jbouder** [reply]
Forgot to mention this earlier. `@metrostar/comet-uswds@3.3.0` is now available.
